docs/articles/documentation/extending-testcafe/custom-reporter/creating-reporter-plugin/helper-methods.md

```diff
@@ -0,0 +1,297 @@
+```
+
+Parameter   | Type    | Description
+----------- | ------- | -------------------------------------------------------
+`str`       | String  | The string you want to break.
+`indentVal` | Integer | The number of spaces to indent a new line.
+`width`     | Integer | The maximum number of characters each line can contain.
+
+**Example**
+
+The following example demonstrates how to break the `title` string onto the next line if the string length is more than 50 symbols.
+
+```js
+var LINE_WIDTH = 50;
+
+reportTestDone (name, errs) {
+    var hasErr    = !!errs.length;
+    var result    = hasErr ? `passed` : `failed`;
+
+    name = `${this.currentFixtureName} - ${name}`;
+
+    var title = `${result} ${name}`;
+
+    title = this.wordWrap(title, 2, LINE_WIDTH);
+
+    this.write(title);
+}
+```
+
+### escapeHtml
+
+Escapes string for use in HTML.
+
+```text
+escapeHtml (str) → this
+```
+
+Parameter   | Type   | Description
+----------- | ------ | ------------------------------
+`str`       | String | The string you want to escape.
+
+**Example**
+
+```js
+reportFixtureStart (name) {
+    this.currentFixtureName = this.escapeHtml(name);
+}
+```
+
+### formatError
+
+Returns the formatted error message string for the specified error.
+
+```text
+formatError (err, prefix = '') → this
+```
+
+Parameter     | Type   | Description
+------------- | ------ | -------------------------------------------------------------------
+`err`         | String | The error message you need to format.
+`prefix = ''` | String | The string that is prepended to the error. By default, it is empty.
+
+**Example**
+
+The following example demonstrates how to add numeration to the errors.
+
+```js
+reportTestDone (name, errs){
+    ...
+    var hasErr = !!errs.length;
+
+    if (hasErr) {
+        errs.forEach((err, idx) => {
+            this.newline()
+                .write(this.formatError(err, `${idx + 1}) `))
+        });
+    }
+}
+```
+
+## Writing Helpers
+
+### newline
+
+Adds a new line to the report.
+
+```text
+newline () → this
+```
+
+**Example**
+
+```js
+reportTaskDone (endTime, passed) {
+    ...
+    var summary = passed === this.testCount ?
+                  `${this.testCount} passed` :
+                  `${this.testCount - passed}/${this.testCount} failed`;
+
+    this.newline()
+        .write(summary)
+        .newline();
+}
+```
+
+### write
+
+Writes the specified text in the report.
+
+```text
+write (text) → this
+```
+
+Parameter | Type   | Description
+--------- | ------ | ---------------------------------
+`text`    | String | The text to output in the report.
+
+**Example**
+
+```js
+reportTaskDone (endTime, passed) {
+    ...
+    var summary = passed === this.testCount ?
+                  `${this.testCount} passed` :
+                  `${this.testCount - passed}/${this.testCount} failed`;
+
+    this.write(summary)
+        .newline();
+}
+```
+
+### useWordWrap
+
+Enables word wrapping for subsequent output.
```

Comment:
I suspect it *toggles* rather than *enables* cause you can disable wrapping as well, can't you?